### PR TITLE
Ticket validation

### PR DESCRIPTION
**Pull request by bnelson7**
still working on routes and trying to understand. syntax is mostly good i think for ticket model couldn't figure out how to create a key of different options for status and priority.

### CODE REVIEW

**Comment by bnelson7** [comment]
fixed routes and ticket model

File: models/Ticket.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        type: Array,
+        default: []
+    },
+    subscribers: {
+        type: Array,
+        default: []
+    },
+    owner: {
+        type: Schema.Types.ObjectId,
+        default: createdDate
```
Comment (haydenlinder) [inline models/Ticket.js]: ref: 'User'

File: models/Ticket.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        type: Schema.Types.ObjectId,
+        ref: 'Ticket'
+    },
+    blocks: {
+        type: Schema.Types.ObjectId,
+        ref: 'Ticket'
+    },
+    startDate: {
+        type: Date,
+        default: Date.now
```
Comment (haydenlinder) [inline models/Ticket.js]: default: undefined

File: models/Ticket.js (hunk 1 above)
Comment (bnelson7) [reply models/Ticket.js]: ok can you see the most recent changes?

File: models/Ticket.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    },
+    endDate: {
+        type: Date,
+        default: Date.now
```
Comment (haydenlinder) [inline models/Ticket.js]: default: 'undefined'

File: models/Ticket.js (hunk 1 above)
Comment (bnelson7) [reply models/Ticket.js]: fixed

File: routes/api/tickets.js [hunk 4]
```diff
@@ -1,6 +1,61 @@
...
+        if (!isValid) {
+            return res.status(400).json(errors);
+        }
+
+        const newTicket = new Ticket({
+            title: req.body.title,
+            body: req.body.body,
+            user: req.user.id
+        });
+
```
Comment (haydenlinder) [inline routes/api/tickets.js]: when you create the ticket, be sure to include all fields a user can input - status, priority, depends on, blocks, start date, and end date

File: routes/api/tickets.js [hunk 5]
```diff
@@ -1,6 +1,52 @@
...
+router.get("/ticket/:id", (req, res) => {
+    Ticket
+        .find({ id: req.params.id })
+        .then(ticket => res.json(ticket))
+        .catch(err => err.status(400).json(err))
+})
+
+router.patch("/ticket/:id", (req, res) => {
+    Ticket
+        .findById({ id: req.params.id })
```
Comment (haydenlinder) [inline routes/api/tickets.js]: Need to update the ticket with the req body

**Pull request merged**